perception/autoware_tensorrt_rtmdet/include/tensorrt_rtmdet/calibrator.hpp

```diff
@@ -0,0 +1,502 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TENSORRT_RTMDET__CALIBRATOR_HPP_
+#define TENSORRT_RTMDET__CALIBRATOR_HPP_
+#include "cuda_utils/cuda_check_error.hpp"
+#include "cuda_utils/cuda_unique_ptr.hpp"
+
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/opencv.hpp>
+
+#include <NvInfer.h>
+#include <assert.h>
+
+#include <algorithm>
+#include <fstream>
+#include <iterator>
+#include <string>
+#include <vector>
+
+namespace autoware::tensorrt_rtmdet
+{
+class ImageStream
+{
+public:
+  ImageStream(
+    int batch_size, nvinfer1::Dims input_dims, const std::vector<std::string> calibration_images)
+  : batch_size_(batch_size),
+    calibration_images_(calibration_images),
+    current_batch_(0),
+    max_batches_(calibration_images.size() / batch_size_),
+    input_dims_(input_dims)
+  {
+    batch_.resize(batch_size_ * input_dims_.d[1] * input_dims_.d[2] * input_dims_.d[3]);
+  }
+
+  int getBatchSize() const { return batch_size_; }
```

Comment:
Thank you for your PR!

I'd like to provide some comments for improvement. Although int is frequently used in Autoware, it is actualy an unsafe type. Please use `uint32_t` instaed.
cf. https://matt.sh/howto-c#:~:text=Writing%20code-,Types,-If%20you%20find

Thank you for your contribution!

Reply:
Hi, really thanks for your support and suggestions.

I chanced most of the types, but I could not change some of them which related with override function from `nvinfer1::IInt8LegacyCalibrator`, `nvinfer1::IInt8EntropyCalibrator2` and `nvinfer1::IInt8MinMaxCalibrator` classes. For example:

- `calibrator.h` -> `line 204`

`int getBatchSize() const noexcept override { return stream_.getBatchSize(); } `

I also putted some minor changes to this commit, I hope it won't be confusing.

related commit: https://github.com/autowarefoundation/autoware.universe/pull/8165/commits/200891589fca0b25681913a292c0f4a88d5ad230